include/quic/opt.hpp

```diff
@@ -13,9 +13,9 @@ namespace oxen::quic::opt
         using Address::Address;
     };
 
-    struct remote_addr : public Address
+    struct remote_addr : public RemoteAddress
     {
-        using Address::Address;
+        using RemoteAddress::RemoteAddress;
     };
```

Comment:
I think we could get rid of these two since I think they are never really optional values, and always the mandatory first argument for `listen` or `connect`.  *Perhaps* we could change that mandatory first argument to be a `opt::{local,remote}_addr`, but I'd be just as fine with leaving it as Address/RemoteAddress and dropping these (since it isn't like the other `opt::` values that go through the almost-named-argument argument handling).

Reply:
I lean towards using the `Address` and `RemoteAddress` types explicitly. I think the idea of using the opt structs as packs is a better use for it anyways